src/main/java/org/kainos/ea/db/JobDao.java

```diff
@@ -43,9 +44,72 @@ public List<Job> getAllJobs(Connection connection) throws SQLException {
             allJobs.add(job);
 
         }
-
         return allJobs;
+    }
+
+    public Job getJobById(Connection connection, int id) throws SQLException {
+        String SQL = "SELECT job_id, job_title, job_spec_summary, capability.name AS capability, job_URL, job_band_name " +
+            "FROM job " +
+            "INNER JOIN job_band USING (job_band_id)" +
+            "INNER JOIN capability USING (capability_id)" +
+            "WHERE job_id = ?";
+
+
+        // Check database connection
+        if (connection == null) {
+            System.err.println("Failed to connect to database");
+            throw new SQLException("Failed to connect to database");
+        }
 
+
+        if(connection.isClosed()){
+            System.err.println("Database connection was closed");
+            throw new SQLException("Database connection was closed");
+        }
+
+        PreparedStatement statement = connection.prepareStatement(SQL);
+        statement.setInt(1,id);
+        ResultSet results = statement.executeQuery();
+
+        // Create a list of Job objects from the results
+
+        if (results.next()) {
+            return new Job(
+                    results.getInt("job_id"),
+                    results.getString("job_title"),
+                    results.getString("capability"),
+                    results.getString("job_spec_summary"),
+                    results.getString("job_url"),
+                    results.getString("job_band_name")
+            );
+        }
+
+        return null;
     }
 
+    public void deleteJob(Connection connection, int id) throws SQLException {
+        String responsibilitiesSQL = "DELETE FROM job_responsibility WHERE job_id = ?";
+        String SQL = "DELETE FROM job WHERE job_id = ?";
+
+        // Check database connection
+        if (connection == null) {
+            System.err.println("Failed to connect to database");
+            throw new SQLException("Failed to connect to database");
+        }
+
+
+        if(connection.isClosed()){
+            System.err.println("Database connection was closed");
+            throw new SQLException("Database connection was closed");
+        }
```

Comment:
this logic should be handles by the database connector

Reply:
Removed